streams/src/main/java/org/apache/kafka/streams/kstream/WindowedSerdes.java

```diff
@@ -50,6 +55,14 @@ public SessionWindowedSerde(final Serde<T> inner) {
         return new TimeWindowedSerde<>(Serdes.serdeFrom(type));
     }
 
+    /**
+     * Construct a {@code TimeWindowedSerde} object to deserialize changelog topic
+     * for the specified inner class type and window size.
+     */
+    static public <T> Serde<Windowed<T>> timeWindowedChangelogSerdeFrom(final Class<T> type, final long windowSize) {
```

Comment:
Note that is users wants to produce a time windowed KTable to a topic they would want to construct a serde with the windowSize, but the `isChangelog` flag would be set to false.

So I'd suggest we have another constructor of `TimeWindowedSerde` which only takes `inner` and `windowSize` and default `isChangelogTopic` to false, and rename this function to `timeWindowedSerdeFrom` which will call that constructor. Then for changelog topics, since in practice it is not very common that users want to fetch from these topics directly we can add a non-constructor `TimeWindowedSerde 
 forChangelog(boolean)` for `TimeWindowedSerde` which will set the boolean flag (again by default it should be false) and then return `this`, and then for those advanced users can then call

```
TimeWindowedSerde serde = timeWindowedSerdeFrom(...).forChangelog(true);
```

Personally I feel it is the leak penetrating API for normal users.